packages/interactive-code-block/src/lib/components/code-mirror/index.tsx

```diff
@@ -150,13 +155,40 @@ const CodeMirror: MemoExoticComponent<
 		}, [languagePlugin, dep, codeMirrorRef.current]);
 
 		useEffect(() => {
-			// return a function to be executed at component unmount
 			return () => {
 				view && view.destroy();
 			};
 		}, [view]);
 
-		return <div ref={codeMirrorRef} className={className} />;
+		useEffect(() => {
+			function keyListener(event: KeyboardEvent) {
+				console.log('key listener');
+				// if (isFocused) {
+				// 	event.preventDefault();
+				// 	event.stopImmediatePropagation();
+				// }
+			}
+			const doc = codeMirrorRef.current!.ownerDocument;
+			console.log({ doc });
+			doc.addEventListener('keydown', keyListener, true);
+			doc.addEventListener('keyup', keyListener, true);
+			doc.addEventListener('keypress', keyListener, true);
+
+			return () => {
+				doc.removeEventListener('keydown', keyListener, true);
+				doc.removeEventListener('keyup', keyListener, true);
+				doc.removeEventListener('keypress', keyListener, true);
+			};
+		}, []);
+
+		return (
+			<div
+				ref={codeMirrorRef}
+				className={className}
+				onKeyDown={stopPropagation}
```

Comment:
Why are we adding listeners both to this element and the owner document? Wouldn't the listeners added to the owner document capture phase catch everything?

Reply:
I changed this file by mistake, it's a different block :-) Let me revert this file

Reply:
Ah, cool :-) This PR a great change, btw.

Reply:
Thanks!